### ISSUE TITLE
DatePicker calendar positioning can be off in Modal

### ISSUE THREAD

**Issue by brunnerh** [carbon-design-system/carbon-components-svelte#953]
The calendar is not correctly offset. The size of the error seems to depend on viewport and/or dialog size.

[REPL](https://svelte.dev/repl/f8571979cc4b46deab2fa7edf069670e?version=3.44.3) (created for a issue #952)

![image](https://user-images.githubusercontent.com/834235/147607571-bbcadb42-5d48-4418-9392-0906ec6d638e.png)

Different viewport:
![image](https://user-images.githubusercontent.com/834235/147608174-b199ec41-2c70-4b41-97aa-e5e02dfeec87.png)

Very small viewport, so the dialog fills the screen:
![image](https://user-images.githubusercontent.com/834235/147608268-aedcc9c6-704e-4d6c-9dcd-1059909d96ae.png)

**Comment by metonym** [comment]
Can you try specifying `flatpickrProps={{ static: true }}` for the calendar to be statically positioned?

```svelte
<DatePicker datePickerType="single" flatpickrProps={{ static: true }}>
  <DatePickerInput />
</DatePicker>
```

**Comment by brunnerh** [comment]
@metonym That seems to fix the problem,, thank you and sorry for not checking the closed issues.

I wonder if the problems that can come up with using `static: true` are actually more frequent than the current default, though.

**Comment by metonym** [comment]
Great point – it's worth considering making `static: true` the default.

**Comment by metonym** [comment]
Closed in favor of #985
